src/main/java/com/gachigang/ontherun/controller/DepartmentController.java

```diff
@@ -28,4 +30,16 @@ public class DepartmentController {
     public Department getDepartmentByUserId(@AuthenticationPrincipal User user) {
         return departmentService.getDepartmentByUserId(user.getId());
     }
+
+    /**
+     * Endpoint for getting the departments of the specific business via id
+     *
+     * @return A list of {@link Department} objects.
+     * @throws {@link NotFoundException} if business with id was not found,
+     * or {@link RuntimeException} if there is no departments for this business
+     */
+    @GetMapping("/{businessId}")
+    public Set<Department> getDepartmentsByBusinessId(@PathVariable Long businessId) {
```

Comment:
Hmmm, we have to return DTO from controller layer, even we must -- because otherway we will get recurcive JSON in responce and a lot of queries in DB

Reply:
Done